common/WhirlyGlobeLib/src/VectorObject.cpp

```diff
@@ -508,19 +506,17 @@ bool VectorObject::pointNearLinear(const Point2d &coord,float maxDistance,ViewSt
     if (!ScreenPointFromGeo(coord, globeView, mapView, coordAdapter, frameSize, modelAndViewMat, modelAndViewMat4d, modelMatFull, modelAndViewNormalMat, &p))
         return false;
 
-    for (ShapeSet::iterator it = shapes.begin();it != shapes.end();++it)
+    for (const auto &shape : shapes)
     {
-        VectorLinearRef linear = std::dynamic_pointer_cast<VectorLinear>(*it);
-        if (linear)
+        if (const auto linear = dynamic_cast<VectorLinear*>(shape.get()))
         {
-            GeoMbr geoMbr = linear->calcGeoMbr();
+            const GeoMbr geoMbr = linear->calcGeoMbr();
             if(geoMbr.inside(GeoCoord(coord.x(),coord.y())))
             {
-                VectorRing pts = linear->pts;
```

Comment:
We were making a copy of the entire vector here

Reply:
Crazy!